### ISSUE TITLE
[1.5.0 rc2 & 1.5.0 rc3 & 1.5.0 rc4] Basic auth is failing 

### ISSUE THREAD

**Issue by deuvarneys** [logstash-plugins/logstash-output-elasticsearch#130]
I have elasticsearch protected behind an nginx server with basic auth. In 1.4.2, I was able to push logs from logstash using the elasticsearch_http output plugin

output {
	elasticsearch_http{ 
		user => "user"
		password => "password"
		host => "host"
		port => "80"
	}
}

However, since 1.5 (rc2 & rc3), I keep getting "401 Authorization Required" from nginx when using both the method described above and 

output {
	elasticsearch{ 
		protocol => "http"
		user => "user"
		password => "password"
		host => "host"
		port => "80"
        }
}

**Comment by jsvd** [comment]
@deuvarneys I've just tried this with

* elasticsearch 1.5.2
* logstash 1.5.2
* nginx 1.9.1

#### nginx conf

```
worker_processes  1;
daemon off;
events { worker_connections  1024; }

http {
    include       mime.types;
    default_type  application/octet-stream;
    sendfile        on;
    keepalive_timeout  65;
    server {
       listen 8080;
       location / {
          proxy_pass http://localhost:9200;
          auth_basic "Restricted";
          auth_basic_user_file htpasswd;
       }
    }
}
```

#### htpasswd file

`htpasswd -c conf/htpasswd test`

and entered `pwd` as password

##### logstash config

```
input { generator { count => 1 } }
output {
  elasticsearch {
    protocol => http
    user => test
    password => pwd
    host => localhost
    port => 8080
  }
}
```

#### elasticsearch

just unpacked a release and did `bin/elasticsearch`

_______

after starting elasticsearch, nginx and logstash I saw 1 event indexed in elasticsearch without issues.

**Comment by deuvarneys** [comment]
Okay, I believe I found the issue. I was able to replicate the settings our production server and found the difference between it and the settings you provided. In our production settings, we are using the "@" special character within the password field. It appears that they special character is supported in Logstash 1.4 but not in Logstash 1.5 which is causing the unauthorized message from nginx. Will you no longer be supporting special characters within the password field for future releases?
